src/transformers/models/ernie4_5_moe/modular_ernie4_5_moe.py

```diff
@@ -96,7 +96,46 @@ def forward(self, hidden_states):
         return hidden_states + self.e_score_correction_bias.squeeze()
 
 
-# FIXME: update the expert
+class Ernie4_5_MoeNaiveMoe(nn.ModuleList):
+    def __init__(self, config):
+        super().__init__()
+        self.top_k = config.num_experts_per_tok
+        self.num_experts = config.num_local_experts
+        for _ in range(self.num_experts):
+            self += [Ernie4_5_MoeMLP(config, config.moe_intermediate_size)]
+
+    def forward(self, hidden_states, e_score_correction_bias, device_type):
```

Comment:
This MoE is different as it has an `e_score_correction_bias` which gets added after the Softmax, in order to select the expert

Rather than splitting the Naive MoE into two, I chose to pass the `e_score_correction_bias` directly to the method.